frontend/components/sections/lead-form.js

```diff
@@ -0,0 +1,64 @@
+import Button from "../elements/button";
```

Comment:
You could consider using a validator library such as yup js & even a form library (formik as it works great with yup) so it looks more like a really world app

Reply:
Alex is right it is easier to use a validator library. Using formik will prevent you from managing the errors when the form is submitted.

However, if you choose to do it manually here is an approach based on what we do in the admin panel, using a reducer is easier but here's the example using a state:


```
const [{ modifiedData, formErrors }, setState] = useState({
    modifiedData: { email: '' },
    formErrors: {},
  });
```